lib/decorators/metadata.js

```diff
@@ -5,16 +5,24 @@ module.exports = settings => {
   const { Establishment, Profile } = settings.models;
   const cache = Cacheable();
 
-  return (c, user) => {
+  return c => {
 
     const promises = [
       c.data.establishmentId && cache.query(Establishment, c.data.establishmentId),
       c.data.subject && cache.query(Profile, c.data.subject),
-      c.data.changedBy && cache.query(Profile, c.data.changedBy)
+      c.data.changedBy && cache.query(Profile, c.data.changedBy),
+      c.activityLog && Promise.all(c.activityLog.map(log => cache.query(Profile, log.changedBy)))
     ];
 
     return Promise.all(promises)
-      .then(([establishment, subject, changedBy]) => {
+      .then(([establishment, subject, changedBy, logChangedBys]) => {
+        let activityLog = [];
+
+        activityLog = c.activityLog && c.activityLog.map(log => {
+          log.changedBy = logChangedBys.find(profile => profile && profile.id === log.changedBy) || {};
+          return log;
```

Comment:
These will be in order, so there's no need for the additional `find`
```js
const activityLog = c.activityLog && c.activityLog.map((log, i) => {
  return { ...log, changedBy: logChangedBys[i] };
});
```

(Note: there's also no need for the `let activityLog`, it can be `const` at the point of assignment)